backend/typescript/rest/userRoutes.ts

```diff
@@ -16,49 +17,70 @@ const userService: IUserService = new UserService();
 /* Get all users, optionally filter by a userId or email query parameter to retrieve a single user */
 userRouter.get("/", async (req, res) => {
   const { userId, email } = req.query;
+  const contentType = req.headers["content-type"];
 
   if (userId && email) {
-    res.status(400).json({ error: "Cannot query by both userId and email." });
+    await setResponseByMimeType(res, 400, contentType, [
+      {
+        error: "Cannot query by both userId and email.",
+      },
+    ]);
     return;
   }
 
   if (!userId && !email) {
     try {
       const users = await userService.getUsers();
-      res.status(200).json(users);
+      await setResponseByMimeType(res, 200, contentType, users);
     } catch (error) {
-      res.status(500).json({ error: error.message });
+      await setResponseByMimeType(res, 500, contentType, [
+        {
+          error: error.message,
+        },
+      ]);
     }
     return;
   }
 
   if (userId) {
     if (typeof userId !== "string") {
-      res
-        .status(400)
-        .json({ error: "userId query parameter must be a string." });
+      await setResponseByMimeType(res, 400, contentType, [
```

Comment:
When a `userId` query param is provided we only return a single user so the CSV option is not needed. Sorry for the extra work!